be/src/exec/vectorized/olap_scan_node.cpp

```diff
@@ -553,6 +573,48 @@ Status OlapScanNode::_capture_tablet_rowsets() {
     return Status::OK();
 }
 
+struct TypeMemoryUsed {
```

Comment:
Looks like this piece of code has been written many times....

Reply:
We are now estimating the size of the actual memory footprint. 1/4 of the memory size during the query is enough. The consume of memory tried in the actual query will be much more (e.g. some storage layer page data overhead). These may take up more than half of the query memory. 
The main purpose of this PR is also to reduce the OOM caused by too many column tasks with too many concurrency scan (such as 600 col+).
we can also use this as a configuration.

Reply:
I have no idea for this. Is there any suggestion?